pymongo/asynchronous/periodic_executor.py

```diff
@@ -27,6 +27,12 @@
 
 _IS_SYNC = False
 
+# The default asyncio loop implementation on Windows
+# has issues with sharing sockets across loops (https://github.com/python/cpython/issues/122240)
+# We explicitly use a different loop implementation here to prevent that issue
+if not _IS_SYNC and sys.platform == "win32":
+    asyncio.set_event_loop_policy(asyncio.WindowsSelectorEventLoopPolicy())  # type: ignore[attr-defined]
```

Comment:
We can't call asyncio.set_event_loop_policy since that has global side effects. Instead we need to explicitly create our on loop, possibly only when the default would be the ProactorEventLoop.